src/common/transformations/src/transformations/common_optimizations/eliminate_unsqueeze_gather.cpp

```diff
@@ -53,57 +55,49 @@ ov::pass::EliminateUnsqueezeGather::EliminateUnsqueezeGather() {
         const auto& m_unsqueeze = patternValue.at(unsqueeze);
         const auto& m_unsqueezeInput = patternValue.at(unsqueezeInput);
 
-        ngraph::copy_runtime_info(m_gather.get_node_shared_ptr(), m_unsqueeze.get_node_shared_ptr());
+        copy_runtime_info(m_gather.get_node_shared_ptr(), m_unsqueeze.get_node_shared_ptr());
         m_gather.replace(m_unsqueezeInput);
 
         return true;
     };
 
-    auto m = std::make_shared<ngraph::pattern::Matcher>(gather, matcher_name);
+    auto m = std::make_shared<Matcher>(gather, matcher_name);
     register_matcher(m, callback);
 }
 
+bool scalar_with_one_consumer(const Output<Node>& out) {
+    return rank_equals(0)(out) && consumers_count(1)(out);
+}
+
 ov::pass::EliminateGatherUnsqueeze::EliminateGatherUnsqueeze() {
```

Comment:
is this multiple unsqueeze case supported in the new version? it looks like it doesn't
![image](https://github.com/openvinotoolkit/openvino/assets/21181928/655e948d-444a-4bc0-a1cc-0807f3c09e7a)
I saw it the real "large" model (codet5-base-sum or dolly-v2-2.8  not so sure which one) , so I believe we have to handle this case, otherwise this is a regression

Reply:
I covered your question in the PR description early on. Repeating it here:

Made `EliminateGatherUnsqueeze` work with Gather -> Binary -> Unsqueeze subgraphs. I removed recent changes in this transformation -- instead of working with many `Unsqueeze`s in this transformation I run `SharedOpOptimization` before it -- and all the `Unsqueeze`s will be optimized into one.

Reply:
could you check it with the codet5-base-sum model?

Reply:
Added test for such case (I was surprised to find out that this logic wasn't covered with tests beforehand)